analyzer/runtime/evm/erc721.go

```diff
@@ -108,20 +107,46 @@ func evmDownloadNFTERC721Metadata(ctx context.Context, logger *log.Logger, sourc
 		)
 		return nil
 	}
+	// 1. Read metadata into interface{} to allow any syntactically correct
+	//    JSON value.
+	var metadataAny interface{}
 	limitedReader := io.LimitReader(rc, MaxMetadataBytes)
-	var metadataBuilder strings.Builder
-	teeReader := io.TeeReader(limitedReader, &metadataBuilder)
-	var metadata ERC721AssetMetadata
-	if err = json.NewDecoder(teeReader).Decode(&metadata); err != nil {
-		logger.Info("error decoding token metadata",
+	if err = json.NewDecoder(limitedReader).Decode(&metadataAny); err != nil {
+		logger.Info("error decoding token metadata as any",
 			"uri", nftData.MetadataURI,
 			"err", err,
 		)
+		if err = rc.Close(); err != nil {
+			return fmt.Errorf("closing metadata reader: %w", err)
+		}
+		return nil
 	}
 	if err = rc.Close(); err != nil {
 		return fmt.Errorf("closing metadata reader: %w", err)
 	}
-	nftData.Metadata = common.Ptr(metadataBuilder.String())
+	// 2. Re-serialize into JSON for database. This is to normalize the
+	//    syntax, in case Go and our database have slightly different opinions
+	//    on what's valid JSON.
```

Comment:
For my understanding: 

This is more hypothetical and was not the source of the errors we were observing, right? Go's json doesn't allow unquoted keys. ([example](https://go.dev/play/p/fl8D2dGIp1a))
So probably the most important change is the new `return nil` in line 122 that demands that metadata be (Go-style) well-formed.

Reply:
right, hypothetical